### ISSUE TITLE
Gromacs to amber conversion - GromacsError: Cannot SETTLE a 4-atom molecule

### ISSUE THREAD

**Issue by aminsagar** [ParmEd/ParmEd#1190]
Dear ParmEd Users/Developers,
I am trying to convert a gromacs topology to amber using DES-amber force field (https://github.com/paulrobustelli/Force-Fields). 
My input script is 
```
gromber topol.top 1584-ions.gro topdir des-amber.ff radii mbondi2
hmassrepartition
outparm 1584-ions-hmr.parm7 1584-ions-hmr.rst7
```
But, I get many warnings 
"ParameterWarning: Overriding DihedralType in DihedralTypeList with same periodicity"
and then an error 
Action gromber failed
        GromacsError: Cannot SETTLE a 4-atom molecule

The tip4pd.itp file looks is as follows
```
[ moleculetype ]
; molname       nrexcl
SOL             1

[ atoms ]
;       at type       res nr  res name  at name       cg nr     charge     mass
1       OW              1       SOL      OW             1       0        16.00000
2       HW              1       SOL     HW1             1       0.5800    1.00800
3       HW              1       SOL     HW2             1       0.5800    1.00800
4       Vrt             1       SOL     MW              1      -1.1600    0.00000

[ settles ]
; i     funct   doh     dhh ---> same as standard tip4p
1       1       0.09572 0.15139

[ virtual_sites3 ]
; Dummy from                    funct   a               b
4       1       2       3       1       0.131937768     0.131937768

[ exclusions ]
1       2       3       4
2       1       3       4
3       1       2       4
4       1       2       3
```
What could be the possible reasons for this error?
I would be really grateful for any suggestions.

Amin.

**Comment by swails** [comment]
I opened a PR in paulrobustelli/Force-Fields#5 that should fix this problem.  SETTLE is an algorithm for 3-constraint systems *specifically*.  Since the `Vrt` atom is assigned atomic number 1, ParmEd doesn't recognize it as a virtual site (but rather thinks it's a hydrogen) and so thinks this residue has 4 atoms (not the 3 required for SETTLE).

I saw several cases where GROMACS ITP files set the atomic number of virtual sites to 0 ([example](https://github.com/gromacs/gromacs/blob/master/share/top/amber99.ff/ffnonbonded.itp#L71-L74), [example](https://github.com/gromacs/gromacs/blob/master/share/top/gromos43a2.ff/ffnonbonded.itp#L49-L53)).  I wasn't exhaustive, but assumed I could rely on atomic numbers being accurate for virtual sites as well as real atoms.
